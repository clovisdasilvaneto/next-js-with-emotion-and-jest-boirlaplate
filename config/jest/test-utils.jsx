// Based in https://testing-library.com/docs/react-testing-library/setup#custom-render

import {render} from '@testing-library/react';
import Providers from 'src/providers/index';

const customRender = (ui, options = {}) =>
  render(ui, {wrapper: Providers, ...options});

// re-export everything
export * from '@testing-library/react';

// override render method
export {customRender as render};
