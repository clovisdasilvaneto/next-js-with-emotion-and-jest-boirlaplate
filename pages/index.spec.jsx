import {render} from 'config/jest/test-utils';
import Home from './index';

describe('Hom Component', () => {
  it('Should render the defualt layout', () => {
    const {getByText, container} = render(<Home />);
    const text = getByText(/^Hey/);

    expect(text).toBeInTheDocument();
    expect(container.innerHTML).toMatchSnapshot();
  });
});
