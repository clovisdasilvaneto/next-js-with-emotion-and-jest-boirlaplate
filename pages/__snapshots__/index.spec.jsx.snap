// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hom Component Should render the defualt layout 1`] = `"<div><p class=\\"css-qm6gfa\\">Hey, I'm using emotion with theme!</p></div>"`;
